File: problems/python/1029.py
def main():
    from sys import stdin, stdout

    def fib(n):
        a, b = 0, 1
        for _ in range(n):
            a, b = b, a + b
        return a

    for _ in range(int(stdin.readline())):
        p = int(stdin.readline())
        x, y = fib(p), fib(p - 1)
        num_calls = (2 * x - 1) + (2 * y - 1)
        stdout.write(f'fib({p}) = {num_calls} calls = {x}\n')
main()

File: problems/c/1029.c
#include <stdio.h>

int chamadas;
int fib(int n);

int main(void)
{
    int f, n, t;
    scanf("%d", &t);

    while (t--)
    {
        chamadas = 0;
        scanf("%d", &n);
        f = fib(n);
        printf("fib(%d) = %d calls = %d\n", n, chamadas - 1, f);
    }

    return 0;
}

int fib(int n)
{
    chamadas++;
    return (n <= 1 ? n : fib(n - 1) + fib(n - 2));
}
